### ISSUE TITLE
Pick from gallery somethimes not works

### ISSUE THREAD

**Issue by agustinsivoplas** [miguelbcr/RxPaparazzo#88]
I am using the library with fragments (support-v4), but for some strange reason sometimes when I click on an image in the gallery closes but nothings happens. I set debug points on subscribe but never executes. I am running Android P with a Google pixel.

**Comment by miguelbcr** [comment]
Hi @agustinsivoplas !
Are you using the sample app?
In that case, I have to test it on Android P , otherwise would you mind to provide the code snippet where you are calling to RxPaparazzo?

> when I click on an image in the gallery closes but nothings happens

Have you seen something in the log output?

**Comment by agustinsivoplas** [comment]
I have tested de sample app with fragments and works fine. In my case the stack is:
Activity -> Fragment A -> Fragment B (at top of stack, this fragment makes the image picker call).
```

RxBus.listen(String::class.java).subscribe { event ->
            if (event != null && activity != null) {
                if (isAdded && !activity!!.isFinishing) {
                    when (event) {
                        Events.EVENT_PICK_PHOTO ->
                            RxPaparazzo.single(this).crop().usingGallery()
                                    .performOnBackOutOnMain()
                                    .subscribe { response ->
                                        if (response.resultCode() == Activity.RESULT_OK) {
                                            response.targetUI().loadImage(response.data())
                                        }
                                    }
                        Events.EVENT_TAKE_PHOTO -> RxPaparazzo.single(this).crop().usingCamera()
                                .performOnBackOutOnMain()
                                .subscribe { response ->
                                    if (response.resultCode() == Activity.RESULT_OK) {
                                        response.targetUI().loadImage(response.data())
                                    }
                                }
                    }
                }
            }
        } 
```

Check the [video](https://drive.google.com/file/d/1DZxIKWYLiLJ8aMznf5ADL64PuTfEGzQq/view?usp=sharing), the first time not works, after  works and after not works. The error is random. I dont see nothing strange in the logcat.

**Comment by miguelbcr** [comment]
yeah, it is weird, maybe I'd put logs on the Fragment's onDestroy() methods in order to observe their lifecycle...

Also I'd try another approach to see how it behaves, for instance:

- Using a dialog directly on the avatar image -> https://github.com/miguelbcr/RxPaparazzo/issues/47#issuecomment-249407892

- Put the pick photo code directly on the avatar image instead of delegating it to the `RxBus`:

``` java
avatar.setOnClickListener { 
	RxPaparazzo.single(this).crop().usingGallery()
        .performOnBackOutOnMain()
        .subscribe { response ->
            if (response.resultCode() == Activity.RESULT_OK) {
                response.targetUI().loadImage(response.data())
            }
        }}
}
``` 

- Using `flatmap` to keep all things in the same stream

``` kotlin
val disposables = CompositeDisposable()
val onRxBusConnectable = RxBus().listen(String::class.java).publish()

fun onCreate() {
	onRxBusConnectable
	        .filter { event -> event != null && activity != null && isAdded && !activity!!.isFinishing }
	        .filter { event == Events.EVENT_PICK_PHOTO }
	        .flatMap { RxPaparazzo.single(this).crop().usingGallery() }
	        .performOnBackOutOnMain()
	        .subscribe { response ->
	            if (response.resultCode() == Activity.RESULT_OK) {
	                response.targetUI().loadImage(response.data())
	            }
	        }

	onRxBusConnectable
	        .filter { event -> event != null && activity != null && isAdded && !activity!!.isFinishing }
	        .filter { event == Events.EVENT_TAKE_PHOTO }
	        .flatMap { RxPaparazzo.single(this).crop().usingCamera() }
	        .performOnBackOutOnMain()
	        .subscribe { response ->
	            if (response.resultCode() == Activity.RESULT_OK) {
	                response.targetUI().loadImage(response.data())
	            }
	        }

	disposables.add(onRxBusConnectable.connect())
}

fun onDestroy() {
    disposables.dispose()
}
``` 

I don't know what's the problem but try to simplify the flow and go step by step 🤞

**Comment by agustinsivoplas** [comment]
Thanks very much! I have discovered the error with your help. My bad. I have reused the bottom sheet dialog so by mistake I had two activities listen the same event EVENT_PICK_PHOTO (the both listeners launch RxPapparazo). 

The stack was Activity A - >Activity B -> Fragment A -> Fragment B (at top of stack, this fragment makes the image picker call).

So Activity A  and Activity B listen the same event EVENT_PICK_PHOTO. That was my mistake. I have delegate the on click event to a simple listener and works.

```
avatar.setOnClickListener { 
   listener?.onGalleryPicked()
}
```

I'm sorry I made you waste your time.

All the best and thanks again!
